exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/TDigestMergedStatistic.java

```diff
@@ -0,0 +1,135 @@
+
+  public TDigestMergedStatistic() {
+    this.tdigestHolder = new HashMap<>();
+    state = State.INIT;
+  }
+
+  @Override
+  public void initialize(String inputName, double samplePercent) {
+    super.initialize(Statistic.TDIGEST_MERGE, inputName, samplePercent);
+    state = State.CONFIG;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public String getInput() {
+    return inputName;
+  }
+
+  @Override
+  public void merge(MapVector input) {
+    // Check the input is a Map Vector
+    assert (input.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    for (ValueVector vv : input) {
+      String colName = vv.getField().getName();
+      TDigest colTdigestHolder = null;
+      if (tdigestHolder.get(colName) != null) {
+        colTdigestHolder = tdigestHolder.get(colName);
+      }
+      NullableVarBinaryVector tdigestVector = (NullableVarBinaryVector) vv;
+      NullableVarBinaryVector.Accessor accessor = tdigestVector.getAccessor();
+
+      try {
+        if (!accessor.isNull(0)) {
+          TDigest other = TDigest.fromBytes(ByteBuffer.wrap(accessor.get(0)));
+          if (colTdigestHolder != null) {
+            colTdigestHolder.add(other);
+            tdigestHolder.put(colName, colTdigestHolder);
+          } else {
+            tdigestHolder.put(colName, other);
+          }
+        }
+      } catch (Exception ex) {
+        //TODO: Catch IOException/CardinalityMergeException
+        //TODO: logger
+      }
+    }
+  }
+
+  public TDigest getStat(String colName) {
+    if (state != State.COMPLETE) {
+      throw new IllegalStateException(String.format("Statistic `%s` has not completed merging statistics",
+          name));
+    }
+    return tdigestHolder.get(colName);
+  }
+
+  @Override
+  public void setOutput(MapVector output) {
+    // Check the input is a Map Vector
+    assert (output.getField().getType().getMinorType() == TypeProtos.MinorType.MAP);
+    // Dependencies have been configured correctly
+    assert (state == State.MERGE);
+    for (ValueVector outMapCol : output) {
+      String colName = outMapCol.getField().getName();
+      TDigest colTdigestHolder = tdigestHolder.get(colName);
+      NullableVarBinaryVector vv = (NullableVarBinaryVector) outMapCol;
+      vv.allocateNewSafe();
+      try {
+        if (colTdigestHolder != null) {
+          int size = colTdigestHolder.smallByteSize();
+          java.nio.ByteBuffer byteBuf = java.nio.ByteBuffer.allocate(size);
+          colTdigestHolder.asSmallBytes(byteBuf);
+          // NOTE: in setting the VV below, we are using the byte[] instead of the ByteBuffer because the
+          // latter was producing incorrect output (after re-reading the data from the VV).  It is
+          // unclear whether the setSafe() api for ByteBuffer has a bug, so to be safe we are using the
+          // byte[] directly which works.
+          vv.getMutator().setSafe(0, byteBuf.array(), 0, byteBuf.array().length);
```

Comment:
Sampling will throw off the histogram values just like NDV. Do we plan to address it now or later?

Reply:
A random sample should still be usable for computing the quantiles.  I have changed the numRowsPerBucket based on your other comment elsewhere about sampling.